### PR TITLE
TINY-8453: Load the editor theme in parallel with other resources

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-8453

Description of Changes:

Changes how the scripts are loaded so that the theme is now loaded in parallel with the other resources (e.g. the plugins, icons and language pack). Previously it'd load the theme first and then load the rest of the resources.

Pre-checks:
* [x] Changelog entry added
* [x] ~Tests have been added (if applicable)~ Existing tests check it loads, but we can't test the order
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Review by TheSpyder — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/init/Render.ts [hunk 1]
```diff
@@ -45,22 +45,15 @@ const loadLanguage = (scriptLoader: ScriptLoader, editor: Editor) => {
   }
 };
 
-const loadTheme = (scriptLoader: ScriptLoader, editor: Editor, suffix: string): Promise<void> => {
+const loadTheme = (scriptLoader: ScriptLoader, editor: Editor, suffix: string): void => {
```
Comment (TheSpyder) [inline modules/tinymce/src/core/main/ts/init/Render.ts:48]: deepscan notes `scriptLoader` is unused

Comment (lnewson) [reply modules/tinymce/src/core/main/ts/init/Render.ts]: Yeah had just seen that myself as well. Have removed the unused argument.

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/init/Render.ts [hunk 2]
```diff
@@ -45,22 +45,15 @@ const loadLanguage = (scriptLoader: ScriptLoader, editor: Editor) => {
...
-      });
-    }
-
-    const waitForTheme = () => ThemeManager.waitFor(theme);
-    return scriptLoader.loadQueue().then(waitForTheme, waitForTheme);
-  } else {
-    return Promise.resolve();
+  if (Type.isString(theme) && !hasSkipLoadPrefix(theme) && !Obj.has(ThemeManager.urls, theme)) {
+    const themeUrl = Options.getThemeUrl(editor);
+    const url = themeUrl ? editor.documentBaseURI.toAbsolute(themeUrl) : 'themes/' + theme + '/theme' + suffix + '.js';
```
Comment (spocke) [inline modules/tinymce/src/core/main/ts/init/Render.ts:53]: While you are doing some cleanup as well would this be better as template literals? The same goes for the plugin one.

**Review by spocke — APPROVED** [review]

**Review by HAFRMO — APPROVED** [review]

File: modules/tinymce/src/core/main/ts/init/Render.ts (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/src/core/main/ts/init/Render.ts]: Done and I did a couple of other minor cleanup things as well.

**Pull request merged**